dat.js

```diff
@@ -0,0 +1,117 @@
+var assert = require('assert')
+var path = require('path')
+var multicb = require('multicb')
+var importFiles = require('./lib/import-files')
+var createNetwork = require('./lib/network')
+var stats = require('./lib/stats')
+
+module.exports = Dat
+
+function Dat (archive, db, opts) {
+  if (!(this instanceof Dat)) return new Dat(archive, db, opts)
+  if (typeof opts === 'undefined') return Dat(archive, null, db)
+  assert.ok(archive, 'archive required')
+  // assert.ok(db, 'database required') // maybe not be required for multidrive...
+  assert.ok(opts.dir, 'opts.directory required')
+
+  this.path = path.resolve(opts.dir)
+  this.options = opts
+
+  this.archive = archive
+  this.db = db
+  this.key = archive.key // only resumed/owned archives will have keys here
+  this.live = archive.live
+  this.owner = archive.owner
```

Comment:
Yep! I force the archive open in [init-archive](https://github.com/datproject/dat-node/blob/multidrive-support/lib/init-archive.js#L33) and set owner to false if the archive is being downloaded. As I say in my comments there, its a bit hacky.